src/integrationtest/java/com/ericsson/gerrit/plugins/eiffel/integrationtest/ServiceIntegrationSteps.java

```diff
@@ -0,0 +1,282 @@
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.http.Header;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.After;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.eiffelcommons.utils.HttpExecutor;
+import com.ericsson.eiffelcommons.utils.HttpRequest;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
+import com.ericsson.eiffelcommons.utils.ResponseEntity;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.RegexMatchFailedException;
+import com.google.gson.JsonObject;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.GetResponse;
+
+import cucumber.api.java.Before;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+
+public class ServiceIntegrationSteps {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceIntegrationSteps.class);
+
+    private final static String RABBITMQ_HOSTNAME = "localhost";
+    private final static int RABBITMQ_PORT = 5672;
+    private final static String RABBITMQ_USERNAME = "myuser";
+    private final static String RABBITMQ_PASSWORD = "myuser";
+    private final static String RABBITMQ_QUEUENAME = "gerrit-test-queue";
+    private final static String RABBITMQ_EXCHANGE_NAME = "eiffel-exchange";
+    private final static String GERRIT_BASE_URL = "http://localhost:8080";
+
+    private String projectName;
+    private String gerritAccountCookie;
+    private String gerritXAuthToken;
+
+    @Before
+    public void setUp() throws Exception {
+        declareQueues();
+        cleanUp();
+        generateGerritAccountCookie();
+        generateGerritXAuthToken();
+    }
+
+    @Given("^a project is created$")
+    public void aProjectIsCreated()
+            throws ClientProtocolException, URISyntaxException, IOException {
+        this.projectName = UUID.randomUUID().toString();
+        ResponseEntity response = createProject(projectName);
+
+        int expected = 201;
+        int actual = response.getStatusCode();
+        assertEquals("Failed to create project", expected, actual);
+    }
+
+    @Given("^the project is configured to send eiffel events with publish url \"([^\"]*)\"$")
+    public void theProjectIsConfiguredToSendEiffelEventsWithPublishUrl(String publishUrl)
+            throws ClientProtocolException, URISyntaxException, IOException {
+        ResponseEntity response = enableEiffelMessaging(publishUrl);
+        int expected = 200;
+        int actual = response.getStatusCode();
+        assertEquals("Failed to configure project", expected, actual);
+    }
+
+    @Given("^a change is created$")
```

Comment:
Is there a reason for not using `@When` here?

Reply:
Well, no specific reason... but now "when" is places instead of "And" for better readability.

Reply:
I think it was just a mistake by us :)